faiss/gpu/GpuResources.cpp

```diff
@@ -153,6 +168,10 @@ cudaStream_t GpuResources::getDefaultStreamCurrentDevice() {
     return getDefaultStream(getCurrentDevice());
 }
 
+raft::handle_t& GpuResources::getRaftHandleCurrentDevice() {
```

Comment:
I was about to start on implementing a resources object (whether in StandardGpuResources or in a separate RaftGpuResources) that would route all Faiss GPU memory allocations to the RAFT memory manager as well. Do you think this is still needed, as I think we discussed this a while ago?

Or maybe it's fine just to have the two memory resources separate.

Reply:
The nice thing about `rmm` (RAPIDS memory manager) is that it allows a user to set a default memory resource for the entire process. I do think it would still be useful if we had an `RmmGpuResources` in addition to the `StandardGpuResource`. The `raft::handle_t` is similar to `GpuResources` in that it's meant to hold on to various expensive resource handles so that they can be reused. The RAFT functions already use the default `rmm` resource for all temporary allocations so if we can have an `RmmGpuResources` right in FAISS then the user can set, for example, a pooled or managed resource using RMM and then both RAFT and FAISS will be using the exact same memory resource. We took a stab at building an `RmmGpuResources` awhile back [here](https://github.com/rapidsai/raft/blob/branch-23.04/cpp/include/raft/spatial/knn/faiss_mr.hpp). Would that be useful?

We would still want to manage a RAFT handle in addition to that though, because the `raft::handle_t` is ultimately what we pass to the RAFT APIs. That said, though, the handle doesn't have to create it's own resources- we can make sure the `raft::handle_t` uses the same resources like the main stream, stream pool, and cublas handle from the `GpuResources` instance.

Reply:
Just a high level question: how does that interact with the pytroch mem manager? People often combine pytorch and Faiss and currently the memory management is disjoint.